lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1402,27 +1403,34 @@ protected function getSelectManyToManyJoinSQL(array $manyToMany)
         return ' INNER JOIN ' . $joinTableName . ' ON ' . implode(' AND ', $conditions);
     }
 
+    public function getMaxBatchedInsertQueries()
+    {
+        $idGenerator    = $this->class->idGenerator;
+        $isPostInsertId = $idGenerator->isPostInsertGenerator();
+
+        if ($isPostInsertId || $this->class->isVersioned) {
+            return 1;
+        }
+
+        return 50;
```

Comment:
Magic number! Why 50? Shouldn't it be configurable?

Reply:
Yes, it should. Please advise how.

Reply:
I would use a constructor argument

Reply:
I agree here. Any fixed number may perform differently depending on the original amount of entities that need to be processed.

Reply:
This information should be set by a new annotation @BatchSize(size = 100) that is configured at the entity level.